### PR TITLE
added options for greatlakes petsc installation

### PR DESCRIPTION
**Pull request by yqliaohk**
## Purpose
Greatlakes seems to requires specifying the blas/lapack lib path in a specific way. Added options that include how to specify that.

## Type of change
- Documentation update

### CODE REVIEW

File: clusterSetup/GreatlakesSetup.rst [hunk 1]
```diff
@@ -33,9 +33,10 @@ home directory on Great Lakes.
    fi
 
    module load python2.7-anaconda/2019.03
-   module load gcc/4.8.5
-   module load openmpi/3.1.4
+   module load gcc/8.2.0
+   module load openmpi/4.0.2
    module load cmake/3.13.2
```
Comment (nwu63) [inline clusterSetup/GreatlakesSetup.rst:38]: I don't actually think we need a newer version of cmake, the version that comes by default is fine (I tested for CGNS), but I guess there is no harm in loading a newer version. I also load the `git` module which provides `git/2.20.1`. No need to change this section if you/@eirikurj don't feel it's relevant, I think this is fine.

File: installInstructions/install3rdPartyPackages.rst [hunk 2]
```diff
@@ -258,6 +258,8 @@ Putting these options together, some complete examples of configuring PETSc are:
       ./configure --with-shared-libraries --download-superlu_dist --download-parmetis=yes --download-metis=yes \
          --with-fortran-bindings=1 --with-debugging=0 --with-scalar-type=real --PETSC_ARCH=$PETSC_ARCH --with-cxx-dialect=C++11
 
+If you are compiling PETSc on Greatlakes, check the :ref:`configure_petsc_gl` for correct options.
```
Comment (nwu63) [inline installInstructions/install3rdPartyPackages.rst:261]: Can you make this into a Note?

File: clusterSetup/GreatlakesSetup.rst (hunk 1 above)
Comment (yqliaohk) [reply clusterSetup/GreatlakesSetup.rst:38]: I don't have a better idea of if we need to load the newer version of cmake or not. But I think loading git/2.20.1 is helpful. Will do that.

File: installInstructions/install3rdPartyPackages.rst (hunk 2 above)
Comment (yqliaohk) [reply installInstructions/install3rdPartyPackages.rst:261]: Sure

File: clusterSetup/GreatlakesSetup.rst (hunk 1 above)
Comment (eirikurj) [reply clusterSetup/GreatlakesSetup.rst:38]: cgnslib recommends a version 3.2.2 and above, so I think we should load the module. I have used 3.13.2 but not tested the more recent 3.17.3.
Same for git, we should be using at least 2.x.x of git.

File: clusterSetup/GreatlakesSetup.rst [hunk 3]
```diff
@@ -149,3 +151,16 @@ To check the estimated starting time for your job, type ``squeue -j <job ID> --s
...
+Greatlakes PETSc installation
+-----------------------------
+
+To build PETSc on Greatlakes, you need to load ``mkl`` first, and specify the blas/lapack lib path in a specific way, which is shown in the example below (``--with-blas-lapack-lib=``):
+
+.. code-block:: bash
+
+   ./configure --with-shared-libraries --download-superlu_dist --download-parmetis=yes --download-metis=yes    --with-fortran-bindings=1 \
+   --with-debugging=0 --with-scalar-type=real --PETSC_ARCH=$PETSC_ARCH --with-cxx-dialect=C++11 --with-mpi-dir=$MPI_HOME COPTFLAGS='-O3 ' CXXOPTFLAGS='-O3 ' FOPTFLAGS='-O3 ' \
+   --CFLAGS='-fPIC' --CXXFLAGS='-fPIC' --FFLAGS='-fPIC'    --with-blas-lapack-lib=" -Wl,--start-group ${MKL_LIB}/libmkl_intel_lp64.a ${MKL_LIB}/libmkl_sequential.a ${MKL_LIB}/libmkl_core.a -Wl,--end-group -lpthread -lm -ldl"
```
Comment (eirikurj) [inline clusterSetup/GreatlakesSetup.rst:166]: Too many spaces between `--FFLAGS='-fPIC'` and `--with-blas-lapack-lib`

File: clusterSetup/GreatlakesSetup.rst [hunk 4]
```diff
@@ -149,3 +151,16 @@ To check the estimated starting time for your job, type ``squeue -j <job ID> --s
...
+
+Greatlakes PETSc installation
+-----------------------------
+
+To build PETSc on Greatlakes, you need to load ``mkl`` first, and specify the blas/lapack lib path in a specific way, which is shown in the example below (``--with-blas-lapack-lib=``):
+
+.. code-block:: bash
+
+   ./configure --with-shared-libraries --download-superlu_dist --download-parmetis=yes --download-metis=yes    --with-fortran-bindings=1 \
+   --with-debugging=0 --with-scalar-type=real --PETSC_ARCH=$PETSC_ARCH --with-cxx-dialect=C++11 --with-mpi-dir=$MPI_HOME COPTFLAGS='-O3 ' CXXOPTFLAGS='-O3 ' FOPTFLAGS='-O3 ' \
```
Comment (eirikurj) [inline clusterSetup/GreatlakesSetup.rst:165]: Remove the space before the `'` at the end of each opt flag list

**Comment by nwu63** [comment]
I just took a look at issue #42, is there anything missing that we want to address in this PR? I'm actually not sure how to use conda properly on GL, I guess we need to do `conda init` and so on to actually have an environment. Is that worth it to get better performance builds of `numpy`?

File: clusterSetup/GreatlakesSetup.rst (hunk 4 above)
Comment (yqliaohk) [reply clusterSetup/GreatlakesSetup.rst:165]: Will do.

**Comment by yqliaohk** [comment]
> I just took a look at issue #42, is there anything missing that we want to address in this PR? I'm actually not sure how to use conda properly on GL, I guess we need to do `conda init` and so on to actually have an environment. Is that worth it to get better performance builds of `numpy`?

Seems the third one in #42 is still missing. Has anyone tried using conda packages on GL? Is it supposed to be better/faster? I never try or test that on clusters. If it's something known to be working well, I'm happy to add relevant things to this PR (or if someone else with experience can do this).

**Comment by nwu63** [comment]
I think for now we can leave it out, and close the issue when we merge this. It's definitely possible but I haven't tried it yet, I just pip install packages (including numpy), and I don't think anyone in our group use conda to install packages on HPC.

**Pull request merged**